var/spack/repos/builtin/packages/root/package.py

```diff
@@ -774,6 +774,14 @@ def add_include_path(dep_name):
             # warnings when building against ROOT
             env.unset("MACOSX_DEPLOYMENT_TARGET")
 
+    def _root_library_path(self):
```

Comment:
I'm now wondering if downstream packages may benefit from this being an `@property`, so we implement the following downstream package code (podio): https://github.com/spack/spack/blob/54ee7d4165824a0e2fd81c91411f71ad50a8106a/var/spack/repos/builtin/packages/podio/package.py#L117-L121
as
```diff
        root_library_path = self.spec["root"].root_library_path
        env.prepend_path(root_library_path, self.spec["podio"].libs.directories[0])
        if "+sio" in self.spec:
            # sio needs to be on LD_LIBRARY_PATH for ROOT to be able to
            # dynamicaly load the python bindings library
            env.prepend_path(root_library_path, self.spec["sio"].libs.directories[0])
```
This would avoid those packages having to do the version checks on ROOT as well.

Reply:
Good idea, @wdconinc.  Changed to a property.  

Note that after this PR is accepted, podio would no longer need to manually adjust `ROOT_LIBRARY_PATH` (and `ROOT_INCLUDE_PATH`) *for itself* as root is a direct dependency of podio.  Manual adjustment wrt sio would still be required:

```diff
        root_library_path = self.spec["root"].root_library_path
-       env.prepend_path(root_library_path, self.spec["podio"].libs.directories[0])
        if "+sio" in self.spec:
            # sio needs to be on LD_LIBRARY_PATH for ROOT to be able to
            # dynamicaly load the python bindings library
            env.prepend_path(root_library_path, self.spec["sio"].libs.directories[0])
```

Reply:
> Manual adjustment wrt sio would still be required

We can cross that bridge when this is merged and we start propagating this through to the dependents :-)